### PR TITLE
`_RenderDecorator.computeDryBaseline` 

### PR DESCRIPTION
**Pull request by LongCatIsLooong**
Depends on https://github.com/flutter/flutter/pull/146363

No test because the consistency of the `computeDryBaseline` implementations is checked at paint time.

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

**Comment by LongCatIsLooong** [comment]
@justinmc could you take a look?

File: packages/flutter/lib/src/material/input_decorator.dart [hunk 1]
```diff
@@ -995,18 +1007,21 @@ class _RenderDecoration extends RenderBox with SlottedContainerRenderObjectMixin
         constraints.maxWidth - (iconWidth + contentPadding.horizontal + prefixIconSize.width + suffixIconSpace),
       );
 
-    // Increase the available width for the label when it is scaled down.
-    final double invertedLabelScale = lerpDouble(1.00, 1 / _kFinalLabelScale, decoration.floatingLabelProgress)!;
+      // Increase the available width for the label when it is scaled down.
+      final double invertedLabelScale = lerpDouble(1.00, 1 / _kFinalLabelScale, decoration.floatingLabelProgress)!;
```
Comment (justinmc) [inline packages/flutter/lib/src/material/input_decorator.dart:1011]: Good catch on the weird indentation.

**Review by justinmc — APPROVED** [review]
LGTM :+1: 

Is the point of this just to make sure we're always calling "dry" operations when possible, or do you have further plans or bugs to fix with this?

**Comment by LongCatIsLooong** [comment]
This enables some bugfixes. Input decorator's linebox height is not correct because prefix/suffix/input should be baseline aligned but we couldn't implement intrinsic height calculation with baseline aligned children.

**Pull request merged**